### ISSUE TITLE
Cannot find LaTeX root file

### ISSUE THREAD

**Issue by FelixBenning** [James-Yu/LaTeX-Workshop#1165]
**Describe the bug**

LaTeX Workshop does not build the document on save of a subfile. 

**To Reproduce**
Create main file `test.tex` with:
```
\documentclass{book}


\begin{document}
\mainmatter
\include{./file}
\end{document}
```
Subfile `file.tex` with:
```
% !TEX root = ./test.tex  

\chapter{Second file}
 test test asdf bla bal asdfa
\endinput
```

**Expected behavior**
Document is being built on save of the subfile.

**Logs**
```
[17:03:52] File watcher: responding to change in d:\Demo\file.tex
[17:03:52] Parsing d:\Demo\file.tex
[17:03:52] Auto-build d:\Demo\file.tex upon save.
[17:03:52] BUILD command invoked.
[17:03:53] Cannot find LaTeX root file.
```

**Screenshots**
![code_2019-02-09_17-13-03](https://user-images.githubusercontent.com/29106652/52523104-03aa9580-2c8e-11e9-8e10-41b743f192e4.png)

**Desktop (please complete the following information):**
 - OS: Windows 10
 - VS Code version: 1.31
 - Extension version: 5.21.0

**Additional context**
In my actual document I also got the error message:
`extension-output-#3 is not a valid LaTeX file.`  but I couldn't reproduce that. The bug first appeared today when the tex symbol in the very left sidebar appeared for the first time. Yesterday the same document worked even without the magic comment. 

EDIT: Now I also sometimes get `ACTIONS command invoked.` after which the Build symbols circles endlessly without result.

**Comment by jlelong** [comment]
Your example works for me on Mac OS. Try removing the `./` in the `include` and also in the magic comment. By the way, the magic command is useless, the extension should automatically that `test` is the root file for `file`.

**Comment by FelixBenning** [comment]
![code_2019-02-09_17-44-20](https://user-images.githubusercontent.com/29106652/52523504-5a19d300-2c92-11e9-8da4-f0d56fdd7289.png)
has no effect (It builds, when changing the main file but not the subfile)

**Comment by jlelong** [comment]
The problem is that the root file is not found for `file.tex`. It should be detected automatically and moreover you have a magic comment. I am a bit puzzled.
Can you post the complete the output from LaTeX Workshop starting from the time when you open the `Demo` folder.

**Comment by FelixBenning** [comment]
```
[17:42:33] Initializing LaTeX Workshop.
[17:42:33] Creating LaTeX Workshop http and websocket server.
[17:42:33] LaTeX Workshop initialized.
[17:42:33] Found root file from active editor: d:\Demo\test.tex
[17:42:33] Root file changed from: undefined. Find all dependencies.
[17:42:33] Instatiating new file watcher for d:\Demo\test.tex
[17:42:33] Parsing d:\Demo\test.tex
[17:42:33] Adding d:\Demo\file.tex to file watcher.
[17:42:33] Parsing d:\Demo\file.tex
[17:42:33] Server created on 127.0.0.1:51233
[17:42:34] Snippet data loaded.
[17:42:34] VIEW command invoked.
[17:42:34] Found root file from active editor: d:\Demo\test.tex
[17:42:34] Root file remains unchanged from: d:\Demo\test.tex.
[17:42:34] Serving PDF file at http://127.0.0.1:51233/viewer.html?file=/pdf:d%25253A%25255CDemo%25255Ctest.pdf
[17:42:34] Open PDF tab for d:\Demo\test.pdf
[17:42:34] LaTeX Workshop version: 5.21.0
[17:42:35] Preview PDF file: d:\Demo\test.pdf
[17:42:52] File watcher: responding to change in d:\Demo\file.tex
[17:42:52] Parsing d:\Demo\file.tex
[17:42:58] File watcher: responding to change in d:\Demo\test.tex
[17:42:58] Parsing d:\Demo\test.tex
[17:42:58] Parsing d:\Demo\file.tex
[17:42:58] Auto-build d:\Demo\test.tex upon save.
[17:42:58] BUILD command invoked.
[17:42:58] Found root file from active editor: d:\Demo\test.tex
[17:42:58] Root file remains unchanged from: d:\Demo\test.tex.
[17:42:58] Building root file: d:\Demo\test.tex
[17:42:58] Build root file d:\Demo\test.tex
[17:42:59] Recipe step 1: latexmk, --max-print-line=10000,-synctex=1,-interaction=nonstopmode,-file-line-error,-pdf,-outdir=d:/Demo,d:/Demo/test
[17:43:02] LaTeX log parsed with 0 messages.
[17:43:02] Recipe of length 1 finished.
[17:43:02] Successfully built d:\Demo\test.tex
[17:43:02] Refresh PDF viewer for d:\Demo\test.pdf
[17:43:02] Cannot convert undefined or null to object
[17:43:03] Preview PDF file: d:\Demo\test.pdf
[17:43:03] Cannot convert undefined or null to object
[17:43:11] File watcher: responding to change in d:\Demo\file.tex
[17:43:11] Parsing d:\Demo\file.tex
[17:43:11] Auto-build d:\Demo\file.tex upon save.
[17:43:11] BUILD command invoked.
[17:43:11] Cannot find LaTeX root file.
[17:43:14] ACTIONS command invoked.
[17:43:14] ACTIONS command invoked.
[17:44:01] File watcher: responding to change in d:\Demo\file.tex
[17:44:01] Parsing d:\Demo\file.tex
[17:44:01] Auto-build d:\Demo\file.tex upon save.
[17:44:01] BUILD command invoked.
[17:44:01] Cannot find LaTeX root file.
```

**Comment by jlelong** [comment]
I really cannot reproduce the problem, so it is virtually impossible for me to understand what is going on. Can someone reproduce the issue and inspect what is going wrong in the [`findRootMagic`](https://github.com/James-Yu/LaTeX-Workshop/blob/master/src/components/manager.ts#L129) and [`findRootDir`](https://github.com/James-Yu/LaTeX-Workshop/blob/master/src/components/manager.ts#L194) functions?

**Comment by FelixBenning** [comment]
If you walk me through it I could try doing that myself. I can also upload the original project if you think that might help.
As I said the TeX symbol below the symbol to the workshop extensions appeared yesterday for the first time if I was not completely oblivious to it before. Which is weird as I did not update either vs code nor did latex workshop have a new version within the last 10 days. So maybe figuring out where that change came from helps?

**Comment by FelixBenning** [comment]
Im trying to reproduce this on my laptop:

- vs code was already installed
- install latex-workshop
- Error: latexmmk: the script engine could not be found
I had this problem before on my PC as I installled miktex not texLive so I am installing perl 5.26.3.2, and latexmk via the miktex console after updating MiKTex and all the latex packages
- LaTeX workshop works fine, even without the magic comment
- The TeX symbol in the left bar is missing, installing the update of vs code:
<img width="960" alt="code_2019-02-10_11-36-15" src="https://user-images.githubusercontent.com/29106652/52532615-5769bd00-2d28-11e9-8066-cc5a861748a2.png">

- TeX symbol on the sidebar is still missing, compiling still works as intended, restarting laptop:
- Opening vs code the build circle appears "loading extensions" (or something like that) after it finished the TeX symbol in the left sidebar appears, but compiling still works as intended
- Trying to break it by opening original project
- Still works... I figured out what the "ACTION command invoked" means though, it appears when clicking on the button with the checkmark/circle symbol and opening the command Palette. 
- Trying to reproduce the error by enabling synctex after build, no success
- Trying again on my PC because I can't trust my eyes:

![code_2019-02-10_11-59-41](https://user-images.githubusercontent.com/29106652/52532874-c0066900-2d2b-11e9-8a14-fdf7809daf93.png)

- Thinking about anything I have done on my PC which I didn't on my laptop, remember that I changed some settings, open user preferences, remove:
```
 "files.exclude": {
        "**/.git": true,
        "**/.svn": true,
        "**/.hg": true,
        "**/CVS": true,
        "**/.DS_Store": true,
        "*/**.aux":true,
        "*.aux": true
    },
```
- compilere works again, but I still get the error:
`[12:02:15] extension-output-#3 is not a valid LaTeX file.`

![code_2019-02-10_12-03-42](https://user-images.githubusercontent.com/29106652/52532925-5dfa3380-2d2c-11e9-8507-a7545d5354aa.png)

TL:DR ommitting .aux files from the file explorer breaks the compiler, one error line still unsolved

**Comment by jlelong** [comment]
The TeX symbol in the left bar opens a specific left panel and appears when the active editor contains a `.tex` file. This TeX left panel has been around since version 5.10.2 at least.

I do not understand what `extension-output-#3` refers to.

Exclude `.aux` files from the file explorer has no impact on the extension. There must be something on your PC. If you can figure out what the problem is, I would be happy to add a specific entry in the [wiki FAQ](https://github.com/James-Yu/LaTeX-Workshop/wiki/FAQ).

**Comment by FelixBenning** [comment]
> Exclude `.aux` files from the file explorer has no impact on the extension.
 
You mean you have tried it and it has no impact for you? Because I can now toggle this problem on and off by pasting and removing:
```
"files.exclude": {
        "**/.git": true,
        "**/.svn": true,
        "**/.hg": true,
        "**/CVS": true,
        "**/.DS_Store": true,
        "*/**.aux":true,
        "*.aux": true
    },
```

in the `settings.json`

And as I said: 
>I changed the setting on Friday and the compiler did not break immediately but only on the next day.
So maybe try restart your computer with the thing in the settings? I really have no idea otherwise. I can't come up with anything else that I have done that could have anything to do with it.

**Comment by jlelong** [comment]
I have played with excluding `.aux` files, restarted vscode several times and I was never able to reproduce your issue.
Anyway, I cannot figure out what `extension-output-#3` refers to. Do you have other files in `d:\Google Drive\Demo\`?

**Comment by FelixBenning** [comment]
Just the screenshots I have taken. Removed them, no effect. I can still toggle this problem with the snippet

**Comment by FelixBenning** [comment]
I tried removing all the `.aux` and other build files until just the two `.tex` files were in the folder. No change.

**Comment by FelixBenning** [comment]
The  `extension-output-#3` have gone away sometime during the removal of the files. Even after the files are restored by a new compilation and one toggle of the snippet, the error didn't come back.

**Comment by jlelong** [comment]
Great! So is everything fixed?

**Comment by FelixBenning** [comment]
It appears to be, yes. Although I still don't really understand what happened

**Comment by jlelong** [comment]
Let me know if you ever figure out what really happened.

**Comment by FelixBenning** [comment]
I will

**Comment by funnym0nk3y** [comment]
I'm having a similar issue. It seems to appear when the current focus is not on a .tex file. E.g. when editing my .gitignore in a divided window parallel to my main.tex file and then press latexmk I get the error. But when bringing focus to my main.tex it works. Maybe that's the reason.

**Comment by jlelong** [comment]
I confirm that the root file is computed on the fly using the currently focused editor. The build function immediately returns if it is not a `.tex` file.

**Comment by funnym0nk3y** [comment]
Is it possible to extend the search for the root file to the whole workspace? Would make the workflow more intuitive. Or at least give a error message in which that behaviour is pointed out? Just my 2 cents as I'm not into the project.

**Comment by jlelong** [comment]
We already search the root file in the whole workspace. If you face an issue with the root file, please fill a new bug report.

**Comment by AnmolMago** [comment]
This error appeared for me when the binaries were not in my path.
